genon/preprocessor/src/util/minio_resource.py

```
@@ -0,0 +1,104 @@
+            try:
+                fcntl.flock(self._fd.fileno(), fcntl.LOCK_EX | fcntl.LOCK_NB)
+                self._fd.seek(0)
+                self._fd.truncate()
+                self._fd.write(f"pid={os.getpid()} acquired_at={time.time()}\n")
+                self._fd.flush()
+                return self
+            except OSError as e:
+                if e.errno not in (errno.EACCES, errno.EAGAIN):
+                    raise
+                if (time.time() - start) >= self.timeout_sec:
+                    raise TimeoutError(f"Timed out acquiring lock: {self.lock_path}")
```

Ruff (0.15.10)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)